frontend/catalyst/third_party/oqd/oqd_database_managers.py

```diff
@@ -0,0 +1,496 @@
+            list of stages in the OQD experiment workflow is ['Loading', 'Trapping',
+            'Initialization', 'Experiment', 'Detection'].
+        process: The process within the given stage of the trapped-ion workflow that the parameter
+            applies to.
+        equation: The equation that describes how the parameter is computed, if applicable.
+        value: The parameter value.
+        unit: The unit associated with the parameter value, if applicable.
+    """
+
+    # TODO: Some parameters, mainly laser frequencies, are expressed as the sum of a nominal
+    # wavelength and a frequency offset, e.g. '493 nm - 7.506 GHz', which is equal to 493.0061 nm.
+    # We need to be able to support this.
+    # Furthermore, some parameters may either have one value or have two simultaneous values, e.g.
+    # 'Doppler cooling laser frequency', which has values '493 nm - 7.506 GHz' and
+    # '493 nm + 4.259 GHz'. We need to be able to support this as well.
+
+    name: str
+    description: str = ""
+    stage: str = ""
+    process: str = ""
+    equation: str = ""
+    value: Union[int, float] = None
+    unit: str = ""
+
+    @classmethod
+    def from_dict(cls, name: str, params: dict) -> "OQDDeviceParameter":
+        """Creates an OQDDeviceParameter object from a dictionary.
+
+        Args:
+            name: The name of the device parameter.
+            params: A dictionary containing the device parameter parameters, typically as parsed
+                from a TOML document.
+
+        Returns:
+            OQDDeviceParameter: The OQDDeviceParameter object.
+        """
+        return cls(
+            name=name,
+            description=params["description"],
+            stage=params["stage"],
+            process=params["process"],
+            equation=params["equation"],
+            value=params["value"],
+            unit=params["unit"],
+        )
+
+
+@dataclass
+class OQDDeviceDatabase:
+    """A database class to represent the properties of an OQD device.
+
+    The properties of the device include hardware specification, parameters relating to the
+    experimental apparatus, and generally any other parameters needed by the compiler. Physical
+    constants, such as the energy levels of the ion(s) being uses, are handled separately.
+
+    Attributes:
+        parameters: A dictionary of OQD device parameters.
+    """
+
+    parameters: dict[str, OQDDeviceParameter] = field(default_factory=dict)
+
+    @classmethod
+    def from_toml(cls, filepath_or_buffer: Union[str, PathLike]):
+        """Loads an OQDDeviceProperties object from a TOML file or string.
+
+        Args:
+            filepath_or_buffer: The path to the TOML file or a TOML document string.
+        """
+        try:
+            if os.path.isfile(filepath_or_buffer):
+                document = _load_toml_from_file(filepath_or_buffer)
+            else:
+                document = _load_toml_from_string(filepath_or_buffer)
```

Comment:
Since this little paragraph comes up quite often, maybe outline it into its own function? This way we can just have one `_load_toml` helper, instead of two helpers with the same purpose. The file-vs-string logic can just be checked in that single helper loader.

Reply:
Maybe even consider putting it into toml utils, since it does not seem specific to oqd?

Reply:
Good call. I've moved these helper functions to `toml_utils` and added a `load_toml` function in 6cfe39a5